src/reactionsystem.jl

```diff
@@ -551,14 +557,19 @@ struct ReactionSystem{V <: NetworkProperties} <:
             (p isa Symbolics.BasicSymbolic) || error("Parameter $p is not a `BasicSymbolic`. This is required.")
         end
 
+        # Filters away any potential obervables from `states` and `spcs`.
+        obs_vars = [obs_eq.lhs for obs_eq in observed]
+        unknowns = filter(state -> !any(isequal(state, obs_var) for obs_var in obs_vars), unknowns)
+        spcs = filter(spc -> !any(isequal(spc, obs_var) for obs_var in obs_vars), spcs)
+
         # unit checks are for ODEs and Reactions only currently
         nonrx_eqs = Equation[eq for eq in eqs if eq isa Equation]
         if checks && isempty(sivs)
             check_variables(unknowns, iv)
             check_parameters(ps, iv)
             nonrx_eqs = Equation[eq for eq in eqs if eq isa Equation]
             !isempty(nonrx_eqs) && check_equations(nonrx_eqs, iv)
-            check_equations(equations(cevs), iv)
+            !isempty(cevs) && check_equations(equations(cevs), iv)
```

Comment:
only check this if there actually are any

Reply:
That should probably (definitely) be the case. I will raise an issue at MTK that they should start doing this (and when they do, we can remove this line).